app/design/adminhtml/default/default/template/page/header.phtml

```diff
@@ -31,21 +31,37 @@
             <input id="global_search" name="query" type="text" class="input-text" value="<?php if(!empty($query)): ?><?php echo $query ?><?php else: ?><?php echo Mage::helper('core')->quoteEscape($defSearch) ?><?php endif ?>" onfocus="if(this.value=='<?php echo Mage::helper('core')->quoteEscape($defSearch, true) ?>')this.value=''; " onblur="if(this.value=='')this.value='<?php echo $defSearch ?>';" />
             <div id="global_search_autocomplete" class="autocomplete"></div>
             <script type="text/javascript">
-                new Ajax.Autocompleter(
```

Comment:
The file https://github.com/OpenMage/magento-lts/blob/ecdf1e1325f30cc6f043447dac5f600990233beb/js/scriptaculous/controls.js#L43-L43 

is all about `autocompleter`. Does it make sense to rewrite `controls.js` with vanilla js?

Reply:
I would say no in the sense that I think that the goal was removing both prototype and scriptaculous and since control.js is part of scriptaculous I would drop it altogether (in a million year when this task is finished hehehe)

Reply:
How about creating a separate file: `js/openmage/ajaxcompleter.js`. It'll make it easier to remove prototype dependency in 3rd-party code. I use `Ajax.Completer` in my projects. If it's possible to retain the functionalities of `js/scriptaculous/controls.js` in `js/openmage/ajaxcompleter.js`, then it'll be very nice.

Reply:
that can be done (although it seems to be adding more work to a huge task) but this PR could still be valid

Reply:
It'll make it easier for this as well:  https://github.com/OpenMage/magento-lts/blob/ecdf1e1325f30cc6f043447dac5f600990233beb/js/varien/js.js#L307

Reply:
the thing is, if we rebuild prototype/scripta in another way, we'll kinda have the same library again 🧐 that's a philosophical question that puzzles me

Reply:
I can't help but note that these js libraries had their advantages. From a few lines of easy-to-understand code to a larger and more difficult-to-analyze code. But it is the painful way to get rid of unmaintained old things.

Reply:
they were super useful, they would still be (if they were maintained) and they would be 1/3 of the size probably